cpp/src/io/orc/stripe_enc.cu

```diff
@@ -618,6 +617,100 @@ inline __device__ void lengths_to_positions(volatile T* vals, uint32_t numvals,
 static const __device__ __constant__ int32_t kTimeScale[10] = {
   1000000000, 100000000, 10000000, 1000000, 100000, 10000, 1000, 100, 10, 1};
 
+template <int block_size, typename Storage>
+static __device__ void encode_null_mask(orcenc_state_s* s,
+                                        bitmask_type const* pushdown_mask,
+                                        Storage& scan_storage,
+                                        int t)
+{
+  if (s->stream.ids[CI_PRESENT] < 0) return;
+
+  auto const column = *s->chunk.column;
+  while (s->present_rows < s->chunk.null_mask_num_rows or s->numvals > 0) {
+    // Number of rows read so far
+    auto present_rows = s->present_rows;
+    // valid_buf capacity is byte per thread in block
+    auto const buf_available_bits = encode_block_size * 8 - s->numvals;
```

Comment:
I don't think it can.
simplified relevant part of the loop:
```
buf_available_bits = encode_block_size * 8 - numvals
nrows = buf_available_bits // at most
numvals += nrows // at most

numvals -= num_encoded // can be zero
```
`nrows + numvals is always <= encode_block_size * 8` always holds

Reply:
Ok, missed that part